Kuroba/app/src/main/java/com/github/adamantcheese/chan/features/gesture_editor/AdjustAndroid10GestureZonesView.kt

```diff
@@ -0,0 +1,178 @@
+        color = context.resources.getColor(R.color.gestures_zone_view_background)
+    }
+    private val addedZonesPaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+        style = Paint.Style.FILL
+        color = context.resources.getColor(R.color.gestures_zone_view_added_zones)
+    }
+    private val currentEditableZonePaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+        style = Paint.Style.FILL
+        color = context.resources.getColor(R.color.gestures_zone_view_current_zone)
+    }
+    private val handlePaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+        style = Paint.Style.FILL
+        color = Color.WHITE
+    }
+
+    @Inject
+    lateinit var exclusionZonesHolder: Android10GesturesExclusionZonesHolder
+
+    init {
+        setWillNotDraw(false)
+        inject(this)
+
+        gestureZoneEditorTouchHandler = GestureZoneEditorTouchHandler(this)
+    }
+
+    fun show(
+            attachSide: AttachSide,
+            orientation: Int,
+            skipZone: ExclusionZone?,
+            measuredWith: Int,
+            measuredHeight: Int
+    ) {
+        this.orientation = orientation
+        this.exclusionZonesHolder.fillZones(addedZones, skipZone)
+
+        currentEditableZone = EditableZone().apply {
+            val editableZoneParams = if (skipZone == null) {
+                null
+            } else {
+                skipZone.checkValid()
+
+                EditableZone.EditableZoneParams(
+                        skipZone.left.toFloat(),
+                        skipZone.top.toFloat(),
+                        (skipZone.right - skipZone.left).toFloat(),
+                        (skipZone.bottom - skipZone.top).toFloat()
+                )
+            }
+
+            init(attachSide, measuredWith, measuredHeight, editableZoneParams)
+        }
+
+        shown = true
+        invalidate()
+    }
+
+    fun hide() {
+        orientation = -1
+        onZoneAddedCallback = null
+        currentEditableZone = null
+        addedZones.clear()
+        shown = false
+    }
+
+    fun onAddZoneButtonClicked() {
+        val editableZone = checkNotNull(currentEditableZone) { "currentEditableZone is null" }
+
+        exclusionZonesHolder.addZone(
+                orientation,
+                editableZone.getCurrentAttachSide(),
+                editableZone.getCurrentZone().asRect()
+        )
+
+        onZoneAddedCallback?.invoke()
+    }
+
+    fun setOnZoneAddedCallback(func: () -> Unit) {
+        this.onZoneAddedCallback = func
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    override fun onTouchEvent(event: MotionEvent): Boolean {
+        check(shown) { "View is not shown but onTouchEvent is called!" }
+
+        if (gestureZoneEditorTouchHandler.onTouchEvent(event)) {
+            return true
+        }
+
+        return super.onTouchEvent(event)
+    }
+
+    override fun onTouchStart(x: Float, y: Float) {
+        val editableZone = checkNotNull(currentEditableZone) { "currentEditableZone is null" }
```

Comment:
You should probably just ignore the touch event if the zone is null.

Reply:
I use checkNotNull often because it does two things at once: ensures that currentEditableZone is not null and creates a new local non-null variable. Otherwise I will have to use "!!" which is considered a bad practice in Kotlin because it will just throw an NPE without any message.